apps/ssr-tests-v9/src/test.ts

```diff
@@ -0,0 +1,90 @@
+import chalk from 'chalk';
+import * as fs from 'fs';
+import * as path from 'path';
+import { Browser } from 'puppeteer';
+
+import { hrToSeconds } from './utils/helpers';
+import { launchBrowser } from './utils/launchBrowser';
+
+class RenderError extends Error {}
+
+export async function runTest(browser: Browser, url: string): Promise<void> {
+  const page = await browser.newPage();
+  await page.setRequestInterception(true);
+
+  let error;
+
+  page.on('console', message => {
+    if (message.type() === 'error') {
+      // Ignoring network errors as we have an interceptor that prevents loading everything except our JS bundle
+      if (!message.text().includes('net::ERR_FAILED')) {
+        error = new RenderError(message.text());
+      }
+    }
+  });
+
+  page.on('request', request => {
+    // Our interceptor allows only our HTML and JS output
+    if (request.url() === url || request.url().endsWith('/out-esm.js')) {
+      return request.continue();
+    }
+
+    return request.abort();
+  });
+
+  page.on('pageerror', err => {
+    error = err;
+  });
+
+  await page.goto(url);
+  await page.close();
+
+  if (error) {
+    throw error;
+  }
+}
+
+async function test(): Promise<void> {
+  const startTime = process.hrtime();
+  console.log('Starting a browser...');
+
+  const browser = await launchBrowser();
+  console.log('Using', await browser.version());
+
+  const htmlPath = path.resolve(__dirname, '..', 'dist', 'index.html');
+
+  if (!fs.existsSync(htmlPath)) {
+    throw new Error('"dist/index.html" does not exist, please run "yarn build" first');
+  }
+
+  const url = `file://${htmlPath}`;
+  console.log(`Using "${url}"`);
+
+  await runTest(browser, url);
+  await browser.close();
+
+  console.log(`Test finished successfully in ${hrToSeconds(process.hrtime(startTime))}`);
+}
+
+test().catch(err => {
+  console.log('');
+  console.log(chalk.bgRed.whiteBright(' @fluentui/ssr-tests-v9 '));
+
+  if (err instanceof RenderError) {
+    console.log(
+      [
+        '  The test failed.',
+        'Please use `$ npx serve dist` or `$ open dist/index.html` to open a HTML page that is used in tests.',
```

Comment:
should the dist/index.html and associated javascript be dumped to pipeline artifacts ?

Reply:
TBH I don't see a big benefit and the state could be easily reproduced locally.